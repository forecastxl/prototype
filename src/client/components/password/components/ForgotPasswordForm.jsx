import React, { PropTypes } from 'react'
import { Field, reduxForm } from 'redux-form'
import { onSubmitActions } from 'redux-form-submit-saga'
import RaisedButton from 'material-ui/RaisedButton'
import { Alert } from '../../alert'
import { TextField } from '../../form'

export function DumbForgotPasswordForm({ handleSubmit, pristine, submitting, error }) {
  return (
    <form onSubmit={handleSubmit}>
      {error && <Alert message={error} type={'error'} />}
      <Field name="email" component={TextField} label="Email" fullWidth />
      <RaisedButton
        type="submit"
        label="Reset password"
        primary
        fullWidth
        disabled={pristine || submitting}
      />
    </form>
  )
}

DumbForgotPasswordForm.propTypes = {
  error: PropTypes.string,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired
}

DumbForgotPasswordForm.defaultProps = {
  error: ''
}

export default reduxForm({
  form: 'forgot-password',
  onSubmit: onSubmitActions(
    'REQUEST_RESET_PASSWORD',
    'REQUEST_RESET_PASSWORD_SUCCESS',
    'REQUEST_RESET_PASSWORD_VALIDATION_FAILURE'
  )
})(DumbForgotPasswordForm)
