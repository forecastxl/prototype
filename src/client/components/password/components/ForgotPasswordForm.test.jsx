import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbForgotPasswordForm } from './ForgotPasswordForm'

describe('<DumbForgotPasswordForm />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <DumbForgotPasswordForm handleSubmit={() => {}} pristine submitting={false} />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
