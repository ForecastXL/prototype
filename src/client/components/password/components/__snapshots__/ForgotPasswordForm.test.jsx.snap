// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbForgotPasswordForm /> enables the submit button when touched and not submitting 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    fullWidth={true}
    label="Email"
    name="email"
  />
  <RaisedButton
    disabled={false}
    fullWidth={true}
    label="Reset password"
    labelPosition="after"
    primary={true}
    secondary={false}
    type="submit"
  />
</form>
`;

exports[`<DumbForgotPasswordForm /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    fullWidth={true}
    label="Email"
    name="email"
  />
  <RaisedButton
    disabled={true}
    fullWidth={true}
    label="Reset password"
    labelPosition="after"
    primary={true}
    secondary={false}
    type="submit"
  />
</form>
`;

exports[`<DumbForgotPasswordForm /> renders errors 1`] = `
<form
  onSubmit={[Function]}
>
  <Alert
    message="Error"
    type="error"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Email"
    name="email"
  />
  <RaisedButton
    disabled={true}
    fullWidth={true}
    label="Reset password"
    labelPosition="after"
    primary={true}
    secondary={false}
    type="submit"
  />
</form>
`;
